pkg/network/dns_parser.go

```diff
@@ -193,3 +199,12 @@ func (*dnsParser) extractIPsInto(alias, domainQueried []byte, records []layers.D
 		}
 	}
 }
+
+func isWantedRecordType(checktype layers.DNSType) bool {
+	for _, t := range recordedQueryTypes {
```

Comment:
Let's use a map here instead. Iterating a slice for every packet is going to be costly.